core/psh/psh.c

```diff
@@ -972,7 +975,125 @@ static void psh_signalstop(int sig)
 }
 
 
-static int psh_run(void)
+static int psh_logincredentget(char *buff, int ispasswd, int maxinlen)
+{
+	int inputlen = -1;
+	char c;
+	struct termios origattr, ttyattr;
+
+	/* Setting terminal attributes */
+	if (tcgetattr(STDIN_FILENO, &origattr) < 0) {
+		fprintf(stderr, "psh: saving tty settings fail\n");
+		return -ENOTTY;
+	}
+	ttyattr = origattr;
+	if (ispasswd)
+		ttyattr.c_lflag &= ~(ECHO);
+	else
+		ttyattr.c_lflag |= (ICANON | ECHO);
+
+	if (tcsetattr(STDIN_FILENO, TCSAFLUSH, &ttyattr) < 0) {
+		fprintf(stderr, "psh: setting tty attributes fail\n");
+		return -ENOTTY;
+	}
+	/* user input */
+	for (;;) {
+		if (inputlen < 0) {
+			if (ispasswd)
+				fprintf(stdout, "Password: ");
```

Comment:
Again, nitpicking... I guess a plain `printf` would look more readable here, but that's not a problem.